tools/hilbert.py

```diff
@@ -0,0 +1,589 @@
+        fn = 'Hilbert.yml'
+        log.warning("Missing input file specification: using default '{}'!".format(fn))
+
+    if (fn is not None) and (df is not None):
+        log.error("Input file specification clashes with the input dump specification: specify a single input source!")
+        exit(1)
+
+    if fn is not None:
+        if not URI(None).validate(fn):
+            log.error("Wrong file specification: '{}'".format(fn))
+            exit(1)
+        log.info("Input file: '{}'".format(fn))
+
+    if df is not None:
+        if not URI(None).validate(fn):
+            log.error("Wrong dump file specification: '{}'".format(df))
+            exit(1)
+        log.info("Input dump file: '{}'".format(df))
+
+    if fn is not None:
+        INPUT_DIRNAME = os.path.abspath(os.path.dirname(fn))
+    else:
+        assert df is not None
+        INPUT_DIRNAME = os.path.abspath(os.path.dirname(df))
+
+    cfg = None
+
+    if fn is not None:
+        log.info("Loading '{}'...".format(fn))
+        try:
+            yml = _load(fn)
+            log.info("Input file is a valid YAML!")
+
+            os.chdir(INPUT_DIRNAME)
+            log.info("Data Validation/Parsing: ")
+
+            cfg = parse_hilbert(yml)
+
+        except:
+            log.exception("ERROR: wrong input file: '{}'!".format(fn))
+            exit(1)
+    else:
+        log.info("Loading dump '{}'...".format(df))
+        try:
+            cfg = pickle_load(df)
+            log.info("Input dump file is valid!")
+        except:
+            log.exception("Wrong input dump file: '{}'!".format(df))
+            exit(1)
+
+    if cfg is None:
+        log.error("Could not get the configuration!")
+        exit(1)
+
+    assert isinstance(cfg, Hilbert)
+    log.info("Configuration is OK!")
+
+    return cfg
+
+
+def output_handler(parser, ctx, args):
+    args = vars(args)
+
+    od = None
+    if 'outputdump' in args:
+        od = args['outputdump']
+        log.debug("Specified output dump file: {}".format(od))
+        assert od is not None
+
+        if URI(None).validate(od):
+            if not PEDANTIC:
+                log.warning("Output dump file: '{}' already exists! Will be overwritten!".format(od))
+            else:
+                log.warning("Output dump file: '{}' already exists! Cannot overwrite it in PEDANTIC mode!".format(od))
+                od = None
+    return od
+
+def cmd_list(parser, context, args, obj):
+    ctx = vars(context)
+    pedantic_handler(parser, ctx, args)
+    cfg = input_handler(parser, ctx, args)
+    return cfg.query(obj)
+
+
+@subcmd('cfg_query', help='Load Configuration and query some part of it')
+def cmd_query(parser, context, args):
+    log.debug("Running '{}'" . format('cfg_query'))
+
+    parser.add_argument('-o', '--object', required=False, default='all',
+            help="specify the object in the config (default: 'all')")
+    parser.add_argument('-od', '--outputdump', default=argparse.SUPPRESS,
```

Comment:
Why should `cfg_query` output a dump? Shouldn't that be within the scope of `cfg_verify`?